### PR TITLE
Update groovy joint workflow, branch list, link to snapshot docs and update badges

### PR DESCRIPTION
**Pull request by jamesfredley**
On readme: updates branch list, links to snapshot docs and updates badges

Updates branches for groovy joint workflow, which was not running on the 4.0.x branch.

Updates groovy  joint workflow to latest from grails-core.  It was still using the version from Groovy 3.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -1,13 +1,25 @@
 # Grails Testing Support
 
-image:https://github.com/grails/grails-testing-support/actions/workflows/gradle.yml/badge.svg["Java CI", link: "https://github.com/grails/grails-testing-support/actions/workflows/gradle.yml"]
+image:https://img.shields.io/badge/Revved%20up%20by-Develocity-06A0CE?logo=Gradle&labelColor=02303A["Revved up by Develocity", link: "https://ge.grails.org/scans"]
+image:https://github.com/grails/grails-testing-support/actions/workflows/gradle.yml/badge.svg?event=push["Java CI", link: "https://github.com/grails/grails-testing-support/actions/workflows/gradle.yml"]
+image:https://github.com/grails/grails-testing-support/actions/workflows/groovy-joint-workflow.yml/badge.svg?event=push["Groovy Joint Validation Build", link: "https://github.com/grails/grails-testing-support/actions/workflows/groovy-joint-workflow.yml"]
```
Comment (matrei) [inline README.adoc]: The links on the badges do not seem to work as expected.

File: README.adoc [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+
+Grails 6 branch is https://github.com/grails/grails-testing-support/tree/3.0.x[3.0.x]
 
 Grails 5 branch is https://github.com/grails/grails-testing-support/tree/2.4.x[2.4.x] which includes support for Groovy 3.
 
+Grails 4.0.x branch is https://github.com/grails/grails-testing-support/tree/2.1.x[2.1.x].
+
+Grails 3.x branch is https://github.com/grails/grails-testing-support/tree/1.1.x[1.1.x].
+
 Please checkout https://github.com/grails/grails-testing-support/releases[release notes] for more information.
```
Comment (matrei) [inline README.adoc]: Is "checkout" right in this context or should it be "check out". (Intellij is complaining)

Comment (jamesfredley) [reply README.adoc]: "Check out" is correct.  I am going to change it to "see", since check out is informal.

File: README.adoc (hunk 1 above)
Comment (jamesfredley) [reply README.adoc]: Sorry, I should have checked this locally instead of just copying the existing one 3 times and updating on github.  All working now.

**Comment by jamesfredley** [comment]
The new groovy joint workflow looks great!

**Review by matrei — APPROVED** [review]
Excellent!

**Review by jdaugherty — APPROVED** [review]

**Pull request merged**